### PR TITLE
feat(data-pipeline): CLAP embeddings on Lance splits at finalize

### LINKED ISSUE #1683: feat(data-pipeline): add CLAP embeddings to Lance datasets during finalize
## Summary

Add an opt-in finalize step that writes a 512-d LAION-CLAP audio-embedding column into each `{train,val,test}.lance` split file, so downstream sound-matching / preset-exploration work can condition on a pretrained audio embedding space — the Lance-native counterpart of #1570 (h5/wds shards).

## Scope

- New `DatasetSpec.compute_clap_embeddings` flag (default `False`, lance-only) propagated through `input_spec.json`.
- New `src/synth_setter/pipeline/data/clap_lance.py`: functional core (`clap_augmented_schema`, `iter_clap_batches`) that appends a `clap` fixed-shape-tensor column from an injected encoder, plus a lazy-`transformers` `load_clap_audio_encoder` shell.
- `finalize_lance` loads the encoder once and threads each split's batches through it as the split files are written.
- Audio is downmixed to mono and resampled to CLAP's 48 kHz before encoding.

## Out of scope

- Wiring the `clap` column into datamodules / training (separate follow-up).
- Text embeddings; audio embeddings only.

Part of #267.

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary

Adds an **opt-in finalize step that writes a 512-d LAION-CLAP audio-embedding column into each `{train,val,test}.lance` split file**, so downstream sound-matching / preset-exploration work can condition on a pretrained audio embedding space. This is the Lance-native counterpart of the h5/wds producer tracked in #1570.

- **New flag** `DatasetSpec.compute_clap_embeddings` (default `False`, **lance-only** — a validator rejects it for hdf5/wds). It serializes into `input_spec.json`, so enabling it is a generate-side config override that finalize reads.
- **New module** `pipeline/data/clap_lance.py` — a functional core that maps a split's `(schema, batches)` to the same stream with a trailing `clap` column:
  - `clap_augmented_schema` appends the `(512,)` float32 tensor field (metadata preserved).
  - `clap_augment_split` / `_iter_clap_record_batches` read each batch's `audio`, downmix to mono (float16→float32), run an **injected** encoder, and guard the output shape to exactly `(num_rows, dim)`.
  - `load_clap_audio_encoder` is the thin shell that builds the real encoder behind a lazy `transformers` import (HF `ClapModel`, resample to CLAP's 48 kHz).
- **`finalize_lance`** loads the encoder once (only when the flag is set) and threads each split's batches through it as the split files are written. When the flag is off, the schema and behavior are byte-for-byte unchanged.
- `transformers>=4.40.0` added to the `torch` dependency group (locked at 5.12.0).

### Why a side column on the split file

The `clap` field lives on the finalized split files (which the datamodule opens directly) rather than the shards: the resharder's virtual layout only carries the core `DATASET_FIELD_NAMES`, and the embedding is additive, so existing Lance readers that project specific columns are unaffected.

## Out of scope (follow-ups)

- Wiring the `clap` column into datamodules / training.
- Text embeddings (audio only).
- A `clap_checkpoint` config field — the checkpoint is a pinned constant, reproducible via the spec's `git_sha`; declined as YAGNI for now.

## Test plan

- `uv run pytest tests/pipeline/data/test_clap_lance.py tests/pipeline/entrypoints/test_finalize_dataset_unit.py tests/pipeline/schemas/test_dataset_spec.py tests/test_finalize_dataset.py` → all green.
- Functional core exercised end-to-end against **real Lance I/O** with an injected fake encoder (no model download): schema append + metadata preservation, mono downmix, sample-rate passthrough, width **and** row-count guards, multi-batch, additive round-trip.
- Finalize branch: flag-on appends `clap` to **every** split (train + val), flag-off loads no model and leaves the legacy schema, the encoder is called at the **shard-metadata sample rate**, and the empty-train guard raises. An e2e leg drives the real `finalize(cfg)` entrypoint and reads the `clap` column back from the uploaded `train.lance`.
- `make format` clean (ruff, ruff-format, pyright, pydoclint, docformatter, codespell, …).

## Verification — real CLAP encoder (manual; needs a model download)

`load_clap_audio_encoder` is the one path not covered by automated tests (checkpoint download + GPU). Manual check:

```bash
uv run python -c "
import numpy as np
from synth_setter.pipeline.data.clap_lance import load_clap_audio_encoder
enc = load_clap_audio_encoder()
out = enc(np.zeros((2, 48000), dtype=np.float32), 48000)
print(out.shape, out.dtype)  # expect (2, 512) float32
"
```

End-to-end against a small finalized Lance dataset: set `compute_clap_embeddings=true` on the generate config, run `synth-setter-finalize-dataset`, and confirm `train.lance` carries a `clap` column.

Closes #1683

### CODE REVIEW

File: src/synth_setter/pipeline/data/clap_lance.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    out_schema = clap_augmented_schema(schema, dim)
+    audio_inner = tuple(schema.field(AUDIO_FIELD).type.shape)
+    audio_index = schema.get_field_index(AUDIO_FIELD)
+    for batch in batches:
+        audio = tensor_chunk_to_numpy(batch.column(audio_index), audio_inner)
+        embeddings = np.asarray(encode(_downmix_to_mono(audio), sample_rate))
+        if embeddings.shape[1] != dim:
+            raise ValueError(f"CLAP encoder produced width {embeddings.shape[1]}, expected {dim}")
+        clap_column = tensor_array(embeddings, np.dtype("float32"), (dim,))
+        yield pa.record_batch([*batch.columns, clap_column], schema=out_schema)
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/synth_setter/pipeline/data/clap_lance.py:100]: Addressed in 903b0601 (then on the branch as the renamed \`_iter_clap_record_batches\`). The guard now validates the full shape rather than just the width:

\`\`\`python
if embeddings.shape != (batch.num_rows, dim):
    raise ValueError(
        f"CLAP encoder produced shape {embeddings.shape}, "
        f"expected ({batch.num_rows}, {dim})"
    )
\`\`\`

A `(D,)` 1-D return or a wrong row count both fail this tuple comparison with a clear `ValueError` before the Arrow tensor is built. `test_clap_augment_split_raises_when_encoder_row_count_differs_from_batch` covers the row-count path and the parametrized width test covers the width path.

File: src/synth_setter/pipeline/data/clap_lance.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        schema, out_schema, batches, encode, sample_rate, dim=dim
+    )
+    return out_schema, augmented
+
+
+def load_clap_audio_encoder(
+    checkpoint: str = DEFAULT_CLAP_CHECKPOINT,
+    device: str | None = None,
+) -> EncodeFn:
+    """Load a CLAP checkpoint and return a mono-batch encode callable.
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/synth_setter/pipeline/data/clap_lance.py:136]: Addressed in f7da7322 — added a fake-backed unit test for the encoder shell. `_install_fake_clap_stack` monkeypatches the lazily-imported `transformers.ClapModel`/`ClapProcessor` and `torchaudio.functional.resample` with in-memory stand-ins (no checkpoint download), and the two new tests assert:

- `test_load_clap_audio_encoder_maps_mono_batch_to_embedding_dim` — the returned callable produces `(B, CLAP_EMBEDDING_DIM)` float32 for a dummy mono batch and does **not** resample at the native 48 kHz rate.
- `test_load_clap_audio_encoder_resamples_when_sample_rate_differs` — the resample branch fires with `(orig_freq, 48000)` when `sample_rate != CLAP_SAMPLE_RATE`.

The module/test docstrings that claimed the shell had no automated coverage were updated to match.
